crates/sui-core/src/gateway_types.rs

```diff
@@ -1173,11 +1176,55 @@ pub struct OwnedObjectRef {
 }
 
 #[derive(Eq, PartialEq, Clone, Debug, Serialize, Deserialize, JsonSchema)]
-#[serde(rename = "Event")]
-// TODO: we need to reconstitute this for non Move events
-pub struct SuiEvent {
-    pub type_: String,
-    pub contents: Vec<u8>,
+#[serde(rename = "Event", rename_all = "camelCase")]
+pub enum SuiEvent {
```

Comment:
what is the long term plan of `gateway_types.rs`? There seem to be a lot of similarities between it and non-gateway types. And we seem to be in the middle of moving away from gateway

Reply:
This should become rpc_types